lib/Runtime/Language/JavascriptOperators.h

```diff
@@ -238,12 +238,12 @@ namespace Js
         template <typename T>
         __forceinline static T* TryFromVar(_In_ RecyclableObject* value)
         {
-            return T::Is(value) ? T::UnsafeFromVar(value) : nullptr;
+            return LegacyVarIs<T>(value) ? UnsafeVarTo<T>(value) : nullptr;
```

Comment:
Why does `TryFromVar` use the legacy behavior? Is that just because you don't know what the behavior is?

Reply:
I was trying very hard to not make any behavior changes. There are a bunch of TryFromVar<JavascriptArray> and a couple of TryFromVar<DynamicObject>, and I wanted them to call into the new methods rather than using VarIs (which now matches the assertions that used to be in FromVar).

---
In reply to: [202405808](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r202405808) [](ancestors = 202405808)

Reply:
This has now been fixed and `LegacyVarIs` is gone. New non-template static methods have replaced `TryFromVar<JavascriptArray>` and `TryFromVar<DynamicObject>`. I used the linker to verify that I didn't miss anything, by building with the declarations of the `LegacyVarIs` specializations present but the definitions absent.

---
In reply to: [202433493](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r202433493) [](ancestors = 202433493,202405808)